src/realtime/websocket/extract-note-id-from-path.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import Optional from 'optional-js';
+
+/**
+ * Extracts the note id from the given realtime URL path.
+ *
+ * @param realtimePath The path from which the note id should be extracted
+ * @return The extracted note id
+ * @throws Error if the given string isn't a valid realtime URL path
+ */
+export function extractNoteIdFromPath(realtimePath: string): string {
+  return Optional.ofNullable(new URL(realtimePath, 'https://example.org'))
```

Comment:
You can't use the URL constructor with only a relative part. It needs either an origin in the path itself or as the second argument. (We also use that in the react client to create absolute links in the renderer.)
But because we actually don't care about the domain or protocol I used "example.org".

Why did I choose to use the URL constructor anyway? Because it knows EVERYTHING about parsing an URL. Every corner case. We need one parameter, but what if we decide to pass other parameters? With this approach we don't have to change the logic here.